pallets/loans/src/entities/pricing/external.rs

```diff
@@ -35,14 +35,21 @@ pub struct ExternalPricing<T: Config> {
 	pub price_id: T::PriceId,
 
 	/// Maximum amount that can be borrowed.
-	pub max_borrow_amount: MaxBorrowAmount<T::Balance>,
+	pub max_borrow_amount: MaxBorrowAmount<T::Rate>,
 
 	/// Reference price used to calculate the interest
-	pub notional: T::Rate,
+	pub notional: T::Balance,
 }
 
 impl<T: Config> ExternalPricing<T> {
 	pub fn validate(&self) -> DispatchResult {
+		if let MaxBorrowAmount::Quantity(quantity) = self.max_borrow_amount {
+			ensure!(
+				quantity.frac().is_zero(),
+				Error::<T>::AmountNotMultipleOfPrice
+			)
```

Comment:
I think the error message should be adapted here ^^

Reply:
Why this? Amount still requires to be multiple of Price right now. 

What is happening here is that if the division between amount/price is a quantity with some decimals, then it means that the price was not multiple of the amount.